sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -543,6 +559,69 @@ private static boolean areAllMappingsEqual(MetaData metaData, String... indices)
         return true;
     }
 
+    @VisibleForTesting
+    static String getRerouteIndex(RerouteAnalyzedStatement statement, Row parameters) throws SQLException {
+        ShardedTable shardedTable = statement.tableInfo();
+        if (shardedTable instanceof DocTableInfo) {
+            DocTableInfo docTableInfo = (DocTableInfo) shardedTable;
+            PartitionName partitionName = AlterTableAnalyzer.createPartitionName(statement.partitionProperties(),
```

Comment:
should we fail with a meaningful error if the table isn't partitioned, but a partition clause is present in the statement?

Reply:
@mfussenegger covered in `createPartitionName` already --> created additional test